web-server/src/components/AnimatedInputWrapper/AnimatedInputWrapper.tsx

```diff
@@ -0,0 +1,119 @@
+import React, { useCallback, useEffect, useState } from "react";
+import { themeColors } from "@/theme/schemes/theme";
+
+const AnimatedInputWrapper = () => {
+  const [isAnimationVisible, setIsAnimationVisible] = useState(true);
+  const [value, setValue] = useState("");
+
+  const addAnimation = useCallback(
+    () => setIsAnimationVisible(true),
+    [setIsAnimationVisible]
+  );
+
+  const handleChange = useCallback(
+    (e: any) => setValue(e.target.value),
+    [setValue]
+  );
+
+  return (
+    <div>
+        {isAnimationVisible && !value && <AnimatedPlaceHolderWrapper />}
+        <input
+          value={value}
+          onChange={handleChange}
+          onBlur={addAnimation}
+          placeholder={!isAnimationVisible ? "Search" : ""}
+          style={{backgroundColor: themeColors.primaryAlt,border: "none",outline: "none", right: "0px"}}
+        />
+    </div>
+    
+  );
+};
+
+const popRepos = ["golang/go", "mozilla/rust", "apple/swift", "oven-sh/bun"];
+
+const AnimatedPlaceHolderWrapper = () => {
+  return (
+    <div
+      style={{
+        position: "absolute",
+        top: "50%",
+        transform: "translate(0%,-50%)",
+        fontSize: "1em",
+        display: "flex",
+        gap: "4px",
+      }}
+    >
+      <span>Search for</span>
+      <AnimatedRepos />
+    </div>
+  );
+};
+
+const AnimatedRepos = () => {
+  const [animationIndex, setAnimationIndex] = useState(0);
+  const [repos, setRepos] = useState(popRepos);
+
+  const increment = () =>
+    setAnimationIndex((prev) => {
+      if (prev >= popRepos.length - 1) return 0;
+      return prev + 1;
+    });
+
+  const rearrangeArray = () => {
+    const beforeIndex = repos.slice(0, 1);
+    const fromIndexOnwards = repos.slice(1);
+    const arr = fromIndexOnwards.concat(beforeIndex);
+    setRepos(arr);
+  };
+  
+  useEffect(() => {
+  const animationInterval = setInterval(increment, 2000);
+  let repoInterval: NodeJS.Timeout | undefined;
+  const timeout = setTimeout(() => {
+    repoInterval = setInterval(rearrangeArray, 2000);
+  }, 500);
+  return () => {
+    clearInterval(animationInterval);
+    if (repoInterval !== undefined) {
+      clearInterval(repoInterval);
+    }
```

Comment:
If you initialise repoInterval with -1 instead of not initialising it with a value, you can skip the check and the type definition having undefined for it.

Reply:
I have added a new increment function which solves this